components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/internal/builder/LogoutRequestBuilder.java

```diff
@@ -100,4 +153,113 @@ public LogoutRequest buildLogoutRequest(String subject, String reason,
+        Issuer issuer = issuerBuilder.buildObject();
+        issuer.setValue(issuerId);
+        logoutReq.setIssuer(issuer);
+
+        NameID nameId = new NameIDBuilder().buildObject();
+        nameId.setFormat(SSOConstants.SAML2_NAME_ID_POLICY);
+        nameId.setValue(subject);
+        logoutReq.setNameID(nameId);
+
+        logoutReq.setReason(reason);
+        logoutReq.setDestination(destination);
+
+        SSOAgentCarbonX509Credential ssoAgentCarbonX509Credential =
+                new SSOAgentCarbonX509Credential(tenantId, tenantDomain);
+        setSignature(logoutReq, SignatureConstants.ALGO_ID_SIGNATURE_RSA,
+                new X509CredentialImpl(ssoAgentCarbonX509Credential));
+
+        return logoutReq;
+    }
+
+    /**
+     * Sign the SAML LogoutRequest message
+     *
+     * @param logoutRequest
+     * @param signatureAlgorithm
+     * @param cred
+     * @return
+     */
+
+
+    public static LogoutRequest setSignature(LogoutRequest logoutRequest, String signatureAlgorithm,
+            X509Credential cred) throws Exception {
+        try {
+            Signature signature = (Signature) buildXMLObject(Signature.DEFAULT_ELEMENT_NAME);
+            signature.setSigningCredential(cred);
+            signature.setSignatureAlgorithm(signatureAlgorithm);
+            signature.setCanonicalizationAlgorithm(SignatureConstants.ALGO_ID_C14N_EXCL_OMIT_COMMENTS);
+
+            try {
+                KeyInfo keyInfo = (KeyInfo) buildXMLObject(KeyInfo.DEFAULT_ELEMENT_NAME);
+                X509Data data = (X509Data) buildXMLObject(X509Data.DEFAULT_ELEMENT_NAME);
+                org.opensaml.xml.signature.X509Certificate cert =
+                        (org.opensaml.xml.signature.X509Certificate) buildXMLObject(
+                                org.opensaml.xml.signature.X509Certificate.DEFAULT_ELEMENT_NAME);
+                String value = Base64.encodeBytes(cred.getEntityCertificate().getEncoded());
+                cert.setValue(value);
+                data.getX509Certificates().add(cert);
+                keyInfo.getX509Datas().add(data);
+                signature.setKeyInfo(keyInfo);
+            } catch (CertificateEncodingException e) {
+                throw new SecurityException("Error getting certificate", e);
+            }
+
+            logoutRequest.setSignature(signature);
+
+            List<Signature> signatureList = new ArrayList<Signature>();
+            signatureList.add(signature);
+
+            // Marshall and Sign
+            MarshallerFactory marshallerFactory =
+                    org.opensaml.xml.Configuration.getMarshallerFactory();
+            Marshaller marshaller = marshallerFactory.getMarshaller(logoutRequest);
+
+            marshaller.marshall(logoutRequest);
+
+            Signer.signObjects(signatureList);
+            return logoutRequest;
+
+        } catch (Exception e) {
+            throw new Exception("Error while signing the Logout Request message", e);
```

Comment:
Improvements added in #849 and #1494 (a3c032a) and #1528 (https://github.com/wso2/carbon-apimgt/commit/c6d6cb237bd7654eda952fb2fbd74863e9899cfc#diff-8c2cf6f45187d7a095bac4cf3acb14c2R228)